flutter/shell/platform/tizen/tizen_renderer_vulkan.cc

```diff
@@ -0,0 +1,972 @@
+        .baseArrayLayer = 0,
+        .layerCount = 1,
+    };
+    vkCmdPipelineBarrier(buffer, VK_PIPELINE_STAGE_COLOR_ATTACHMENT_OUTPUT_BIT,
+                         VK_PIPELINE_STAGE_BOTTOM_OF_PIPE_BIT, 0, 0, nullptr, 0,
+                         nullptr, 1, &barrier);
+    if (vkEndCommandBuffer(buffer) != VK_SUCCESS) {
+      FT_LOG(Error) << "Could not end command buffer!";
+      return false;
+    }
+  }
+  return true;
+}
+
+bool TizenRendererVulkan::RecreateSwapChain() {
+  vkDeviceWaitIdle(logical_device_);
+  resize_pending_ = false;
+  for (size_t i = 0; i < present_transition_buffers_.size(); ++i) {
+    vkFreeCommandBuffers(logical_device_, swapchain_command_pool_, 1,
+                         &present_transition_buffers_[i]);
+  }
+  DestroyCommandPool();
+  if (!CreateCommandPool()) {
+    FT_LOG(Error) << "Fail to create command pool!";
+    return false;
+  }
+  if (!InitializeSwapchain()) {
+    FT_LOG(Error) << "Fail to create swapchain!";
+    return false;
+  }
+  return true;
+}
+
+void TizenRendererVulkan::DestroySurface() {
+  if (surface_) {
+    vkDestroySurfaceKHR(instance_, surface_, nullptr);
+  }
+}
+
+void TizenRendererVulkan::ResizeSurface(int32_t width, int32_t height) {
+  if (width_ != width || height_ != height) {
+    width_ = width;
+    height_ = height;
+    resize_pending_ = true;
+  }
+}
+uint32_t TizenRendererVulkan::GetVersion() {
+  return VK_API_VERSION_1_1;
+}
+
+FlutterVulkanInstanceHandle TizenRendererVulkan::GetInstanceHandle() {
+  return instance_;
+}
+
+FlutterVulkanQueueHandle TizenRendererVulkan::GetQueueHandle() {
+  return graphics_queue_;
+}
+
+FlutterVulkanPhysicalDeviceHandle
+TizenRendererVulkan::GetPhysicalDeviceHandle() {
+  return physical_device_;
+}
+
+FlutterVulkanDeviceHandle TizenRendererVulkan::GetDeviceHandle() {
+  return logical_device_;
+}
+
+uint32_t TizenRendererVulkan::GetQueueIndex() {
+  return graphics_queue_family_index_;
+}
+
+const char** TizenRendererVulkan::GetEnabledInstanceExtensions() {
+  return enabled_instance_extensions_.data();
+}
+
+const char** TizenRendererVulkan::GetEnabledDeviceExtensions() {
+  return enabled_device_extensions_.data();
+}
+
+void* TizenRendererVulkan::GetInstanceProcAddress(
+    FlutterVulkanInstanceHandle instance,
+    const char* name) {
+  return reinterpret_cast<void*>(
+      vkGetInstanceProcAddr(reinterpret_cast<VkInstance>(instance), name));
+}
+
+FlutterVulkanImage TizenRendererVulkan::GetNextImage(
+    const FlutterFrameInfo* frameInfo) {
+  if (resize_pending_) {
+    RecreateSwapChain();
+  }
+  VkResult result;
+  do {
+    result = vkAcquireNextImageKHR(logical_device_, swapchain_, UINT64_MAX,
+                                   VK_NULL_HANDLE, image_ready_fence_,
+                                   &last_image_index_);
+    if (result == VK_ERROR_OUT_OF_DATE_KHR) {
+      RecreateSwapChain();
+    } else if (result == VK_SUBOPTIMAL_KHR) {
+      break;
+    }
+  } while (result != VK_SUCCESS);
```

Comment:
If `VK_SUCCESS` doesn't returned, this loop will loop infinitely unless `VK_SUBOPTIMAL_KHR` returned.
It seems like you need to add handling for `VK_TIMEOUT` or a condition that calls `vkAcquireNextImageKHR()` again.
(I don't know the details of how vkAcquireNextImageKHR() works, so if this comment is invalid, please ignore it.)